### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by Neilkot**

### CODE REVIEW

File: src/main/java/com/file/processor/model/ProcessingStatus.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+public enum ProcessingStatus {
+  UPLOADING("Uploading"), UPLOADED("Uploaded"), PROCESSING("Processing"), FAILURE("Failure"), DONE("Done");
+
+  private static Map<String, ProcessingStatus> valuesToItem = new HashMap<>();
```
Comment (Allugard) [inline src/main/java/com/file/processor/model/ProcessingStatus.java:14]: why do we need that?

File: src/main/java/com/file/processor/service/AsyncFileProcessingService.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      FileModel savedFileModel = saveFileToDB(fileJob.getFileModel());
+      enrichWithStatusId(fileJob.getId(), savedFileModel);
+
+      changeStatus(fileJob.getId(), ProcessingStatus.UPLOADED);
+      changeStatus(fileJob.getId(), ProcessingStatus.PROCESSING);
+      performJob(fileJob);
+
+      changeStatus(fileJob.getId(), ProcessingStatus.DONE);
+    } catch (Exception e) {
+      if (fileJob.getNumOfRetries() <= 2) {
```
Comment (Allugard) [inline src/main/java/com/file/processor/service/AsyncFileProcessingService.java:43]: let's extract this const to config

File: src/main/java/com/file/processor/service/AsyncFileProcessingService.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  @Async("asyncExecutor")
+  public void addJob(FileJob fileJob) {
+    try {
+      validateFile(fileJob.getFileModel());
+      FileModel savedFileModel = saveFileToDB(fileJob.getFileModel());
+      enrichWithStatusId(fileJob.getId(), savedFileModel);
+
+      changeStatus(fileJob.getId(), ProcessingStatus.UPLOADED);
+      changeStatus(fileJob.getId(), ProcessingStatus.PROCESSING);
```
Comment (Allugard) [inline src/main/java/com/file/processor/service/AsyncFileProcessingService.java:38]: why status is changing 2 times?

File: src/main/java/com/file/processor/model/ProcessingStatus.java (hunk 1 above)
Comment (Neilkot) [reply src/main/java/com/file/processor/model/ProcessingStatus.java:14]: Do you mean Map "valuesToItem" and all other methods related to it or enum ProcessingStatus?

Comment (Allugard) [reply src/main/java/com/file/processor/model/ProcessingStatus.java:14]: only map

**Pull request closed without merging**